### PR TITLE
Proper CA-style shaders for 3K and WH3

### PR DESCRIPTION
**Pull request by mr-phazer**
# CA-style WH3 and 3K metal-roughness shader

- PBR implementation _**very**_ close to the CA one, for WH3 and 3K. 
- models viewed in Asset Editor will look very close to in-game
- Replaces existing "generic" metal roughness shader
- CA-style shaders covering the other games will follow soon.

### CODE REVIEW

File: GameWorld/ContentProject/Content/Content.mgcb [hunk 1]
```diff
@@ -44,10 +44,16 @@
...
+#begin Shaders/Pbr/MetalRoughness/CA_MetalRoughness_main.fx
 /importer:EffectImporter
 /processor:EffectProcessor
 /processorParam:DebugMode=Auto
+/build:Shaders/Pbr/MetalRoughness/CA_MetalRoughness_main.fx
+
+#begin Shaders/Pbr/MetalRoughness/MetalRoughness_main.fx
+/importer:EffectImporter
+/processor:EffectProcessor
+/processorParam:DebugMode=Debug
```
Comment (donkeyProgramming) [inline GameWorld/ContentProject/Content/Content.mgcb:56]: Should they be in debug mode? This is not how it was before it seems

File: GameWorld/ContentProject/Content/Content.mgcb [hunk 2]
```diff
@@ -59,7 +65,7 @@
 #begin Shaders/TexturePreview.fx
 /importer:EffectImporter
 /processor:EffectProcessor
-/processorParam:DebugMode=Auto
+/processorParam:DebugMode=Debug
```
Comment (donkeyProgramming) [inline GameWorld/ContentProject/Content/Content.mgcb:68]: Should they be in debug mode? This is not how it was before it seems

File: GameWorld/ContentProject/Content/Shaders/Pbr/Capabilites/Tint.hlsli [hunk 3]
```diff
@@ -1,25 +1,34 @@
-bool        CapabilityFlag_ApplyTint = false;
+#include "../helpers/mathfunctions.hlsli"
 
-bool        Tint_UseFactionColours = false;
-float3      Tint_FactionsColours[3] = { float3(1, 0, 0), float3(0, 1, 0), float3(0, 0, 1)};
-float4      Tint_Mask = float4(0, 0, 0, 0);
-float3      Tint_TintColour = float3(1, 1, 1);
-float       Tint_TintVariation = 0;
+// Input parameters 
+bool CapabilityFlag_ApplyTinting = true; // PHAZER: dx11 global non-static consts are = 0, so values are not used
```
Comment (donkeyProgramming) [inline GameWorld/ContentProject/Content/Shaders/Pbr/Capabilites/Tint.hlsli:4]: Avoid putting your name in the codebase like this

File: GameWorld/ContentProject/Content/Shaders/Pbr/MetalRoughness/MetalRoughness_main.fx [hunk 4]
```diff
@@ -56,7 +56,7 @@ GBufferMaterial GetMaterial(in PixelInputType input)
     
     if (UseMask)
     {
-        material.maskValue = MaskTexture.Sample(SampleType, texCord).a;
+        material.maskValue = MaskTexture.Sample(SampleType, texCord);
```
Comment (donkeyProgramming) [inline GameWorld/ContentProject/Content/Shaders/Pbr/MetalRoughness/MetalRoughness_main.fx:59]: Is this correct?

File: GameWorld/View3D/Services/ResourceLibary.cs [hunk 5]
```diff
@@ -58,15 +58,15 @@ public void Initialize(GraphicsDevice graphicsDevice, ContentManager content)
                 CommonSpriteBatch = new SpriteBatch(_graphicsDevice);
 
                 // Load default resources
-                var mr = LoadEffect("Shaders\\Pbr\\MetalRoughness\\MetalRoughness_main", ShaderTypes.Pbs_MetalRough);
+                var mr = LoadEffect("Shaders\\Pbr\\MetalRoughness\\CA_MetalRoughness_main", ShaderTypes.Pbs_MetalRough);
```
Comment (donkeyProgramming) [inline GameWorld/View3D/Services/ResourceLibary.cs:61]: Please dont name the file CA, just remove the prefix as it adds nothing in this context

File: GameWorld/View3D/Services/ResourceLibary.cs [hunk 6]
```diff
@@ -58,15 +58,15 @@ public void Initialize(GraphicsDevice graphicsDevice, ContentManager content)
...
                 var sg = LoadEffect("Shaders\\Pbr\\SpecGloss\\SpecGloss_main", ShaderTypes.Pbr_SpecGloss);
                 LoadEffect("Shaders\\Geometry\\BasicShader", ShaderTypes.BasicEffect);
                 LoadEffect("Shaders\\TexturePreview", ShaderTypes.TexturePreview);
                 LoadEffect("Shaders\\LineShader", ShaderTypes.Line);
                 DefaultFont = _content.Load<SpriteFont>("Fonts//DefaultFont");
 
-                _pbrDiffuse = _content.Load<TextureCube>("textures\\phazer\\DIFFUSE_IRRADIANCE_edited_kloppenheim_06_128x128");
-                _pbrSpecular = _content.Load<TextureCube>("textures\\phazer\\SPECULAR_RADIANCE_edited_kloppenheim_06_512x512");
+                _pbrDiffuse = _content.Load<TextureCube>("textures\\phazer\\SPECULAR_RADIANCE_edited_kloppenheim_06_512x512");
```
Comment (donkeyProgramming) [inline GameWorld/View3D/Services/ResourceLibary.cs:68]: Specular goes into defuse and the other way around. Is this intentional? SHould not textures be renamed if this is the case?

**Review by donkeyProgramming — CHANGES_REQUESTED** [review]

File: GameWorld/ContentProject/Content/Shaders/Pbr/MetalRoughness/MetalRoughness_main.fx (hunk 4 above)
Comment (mr-phazer) [reply GameWorld/ContentProject/Content/Shaders/Pbr/MetalRoughness/MetalRoughness_main.fx:59]: Yes, it is fact correct, 
as Mask Data official are stored in R,B,G,A so you do need to real the real.
and `material.maskValue ` is `float4`

File: GameWorld/View3D/Services/ResourceLibary.cs (hunk 5 above)
Comment (mr-phazer) [reply GameWorld/View3D/Services/ResourceLibary.cs:61]: I do think shader names, for games, should reflect is games it is for

**Pull request closed without merging**